### PR TITLE
[WIP] Slip32

### PR DESCRIPTION
**Pull request by anler**
Export/import keys to [Slip32](https://github.com/satoshilabs/slips/blob/master/slip-0032.md) format.

I'm stuck in: see TODO comment in `wallet/src/slip32.rs` file

### CODE REVIEW

File: wallet/src/slip32.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    // 32 bytes for chain code
+    {
+        let start = 1 + depth * 4;
+        let end = start + 32;
+        buffer.splice(start..end, extkey.chain_code.iter().cloned());
+    }
+
+    // 33 bytes for key
+    //
+    // TODO: Find a way to get access to secp256k1::SecretKey bytes,
```
Comment (Tommytrg) [inline wallet/src/slip32.rs]: In https://github.com/rust-bitcoin/rust-secp256k1/issues/75#issuecomment-456607681 they answer how to get the bytes out from a secret key. 

> You should be able to use [..] to get the bytes out of a secret key no problem. See, eg, this line in rust-lightning: https://github.com/rust-bitcoin/rust-lightning/blob/master/src/ln/chan_utils.rs#L80

File: wallet/Cargo.toml [hunk 2]
```diff
@@ -6,5 +6,11 @@ version = "0.1.0"
 workspace = ".."
 
 [dependencies]
-bip39 = "*"
+bip39 = "0.6.0-beta.1"
+failure = "*"
```
Comment (Tommytrg) [inline wallet/Cargo.toml]:
```suggestion
failure = "0.1.5"
```

File: wallet/Cargo.toml [hunk 3]
```diff
@@ -6,5 +6,11 @@ version = "0.1.0"
...
+hmac = "*"
```
Comment (Tommytrg) [inline wallet/Cargo.toml]:
```suggestion
hmac = "0.7.0"
```

File: wallet/Cargo.toml [hunk 4]
```diff
@@ -6,5 +6,11 @@ version = "0.1.0"
...
+secp256k1 = "*"
```
Comment (Tommytrg) [inline wallet/Cargo.toml]:
```suggestion
secp256k1 = "0.12.2"
```

File: wallet/Cargo.toml [hunk 5]
```diff
@@ -6,5 +6,11 @@ version = "0.1.0"
...
+sha2 = "*"
```
Comment (Tommytrg) [inline wallet/Cargo.toml]:
```suggestion
sha2 = "0.8.0"
```

File: wallet/Cargo.toml [hunk 6]
```diff
@@ -6,5 +6,11 @@ version = "0.1.0"
...
+bech32 = "*"
```
Comment (Tommytrg) [inline wallet/Cargo.toml]:
```suggestion
bech32 = "0.6.0"
```

File: wallet/Cargo.toml [hunk 7]
```diff
@@ -6,5 +6,11 @@ version = "0.1.0"
...
 tiny-hderive = "*"
+endianrw = "*"
```
Comment (Tommytrg) [inline wallet/Cargo.toml]:
```suggestion
endianrw = "0.2.2"
```

File: wallet/Cargo.toml [hunk 8]
```diff
@@ -6,5 +6,11 @@ version = "0.1.0"
...
 
 [dependencies]
-bip39 = "*"
+bip39 = "0.6.0-beta.1"
+failure = "*"
+hmac = "*"
+secp256k1 = "*"
+sha2 = "*"
+bech32 = "*"
 tiny-hderive = "*"
```
Comment (Tommytrg) [inline wallet/Cargo.toml]:
```suggestion
tiny-hderive = "0.1.0"
```

File: wallet/src/key.rs [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+    pub fn new(seed: S) -> Self {
+        Self {
+            key: DEFAULT_HMAC_KEY,
+            seed: seed,
+        }
+    }
+
+    /// Use the given key as the HMAC key
+    pub fn with_key(mut self, key: &'a str) -> Self {
+        self.key = key;
```
Comment (Tommytrg) [inline wallet/src/key.rs]:
```suggestion
        self.key = key;
        
```

**Comment by lrubiorod** [comment]
It is included in #432

**Pull request closed without merging**